### PR TITLE
Rename PropertyFileGraph to PropertyGraph

### LINKED ISSUE #54: Rename PropertyFileGraph,PropertyGraph to PropertyGraph,TypedPropertyGraph
https://github.com/KatanaGraph/katana/pull/53#issuecomment-768379670

### PR DESCRIPTION
**Pull request by ddn0**
I blindly changed the python code, let me know if I did it wrong.

Closes #54

### CODE REVIEW

**Comment by ddn0** [comment]
Oh I was worried about this:
```

#
# Python Property graph
#
cdef class PropertyGraph:
    ^
------------------------------------------------------------

property_graph.pxd:13:5: 'PropertyGraph' redeclared 

Error compiling Cython file:
------------------------------------------------------------
...
        shared_ptr[CUInt64Array] out_indices
        shared_ptr[CUInt32Array] out_dests
        uint64_t num_nodes()
        uint64_t num_edges()

    cppclass PropertyGraph:
   ^
------------------------------------------------------------

/home/runner/miniconda/conda-bld/katana-python_1611969238587/work/python/katana/../katana/cpp/libgalois/graphs/Graph.pxd:110:4: Previous declaration is here

Error compiling Cython file:
------------------------------------------------------------
...
#
# Python Property graph
#
cdef class PropertyGraph:
    cdef:
        shared_ptr[PropertyGraph] underlying
                  ^
------------------------------------------------------------

property_graph.pxd:15:19: Python object type 'PropertyGraph' cannot be used as a template argument

Error compiling Cython file:
------------------------------------------------------------
...
from pyarrow.lib cimport Schema

#
# Python Property graph
#
cdef class PropertyGraph:
    ^
------------------------------------------------------------

property_graph.pxd:13:5: C class 'PropertyGraph' is declared but not defined
```
@arthurp  What's a good way to distinguish the python class PropertyGraph from `katana::PropertyGraph`?

**Comment by arthurp** [comment]
> What's a good way to distinguish the python class PropertyGraph from `katana::PropertyGraph`?

I prefix the C++ class with `_` pyarrow prefixes it with `C`. Either is fine with me. The way to make this work is by using the Cython feature that allows you to specify a C level name for a Cython name.

```Cython
cppclass _PropertyGraph "PropertyGraph":
    ...
```

I have noticed weird interactions between namespaces and this feature. So you may need to provide the namespace in the quoted name `"katana::PropertyGraph"`.

**Comment by ddn0** [comment]
> I prefix the C++ class with _ pyarrow prefixes it with C. Either is fine with me. The way to make this work is by using the Cython feature that allows you to specify a C level name for a Cython name.

Without knowing more, the "C" prefix convention seems more obvious to me, as one may use `_` to denote a variety of things, while "C" seems specific to the case of "this is the underlying C class."

I normally associate the prefix `_` in Python for hidden/internal/not-for-everyone.

Is the intended state that most (end) users would not need to reference `_` prefixed functions, and we intend to provide Pythonic wrappers for everything?

**Comment by arthurp** [comment]
> Is the intended state that most (end) users would not need to reference `_` prefixed functions, and we intend to provide Pythonic wrappers for everything?

Yes. In fact `cdef` functions and `cppclass`es are not callable from Python without an additional wrapper. The `cdef extern` blocks are basically the C++ header restated in a form Cython can understand. No Python user will ever use it. The only people who might use it are Cython programmers who want to integrate with Katana in both C++ and Python.

I agree that `C` is a better prefix. I kind of regret using `_` (I was just thinking "internal", but internal is not the most important thing about the name). I guess I'll change it when I clean things up next.

File: docs/GraphOperationAPIs.md [hunk 1]
```diff
@@ -210,7 +210,7 @@ cdef extern from "[operation header]" namespace "[namespace]" nogil:
             [...]
 
         _[Operation]Plan()
-        _[Operation]Plan(const PropertyFileGraph * pfg)
```
Comment (arthurp) [inline docs/GraphOperationAPIs.md:213]: In the Cython code in this documentation it needs to be `_PropertyGraph*` not `PropertyGraph*`. So where ever it was `PropertyFileGraph` use the `_`.

**Review by arthurp — APPROVED** [review]
LGTM. Just the one commented issue in the documentation.

I reviewed the Python and Cython code fully. I scanned the C++ code.

File: docs/GraphOperationAPIs.md [hunk 2]
```diff
@@ -132,7 +132,7 @@ It returns `katana::ErrorCode::AssertionFailed` if the output is not as expected
 
 ```c++
 KATANA_EXPORT Result<void> [Operation]AssertValid(
-    PropertyFileGraph* pfg, [... functional arguments ...]);
```
Comment (arthurp) [inline docs/GraphOperationAPIs.md:135]: This is C++ not Cython. ;-) Same on line 161

File: libgalois/include/katana/TypedPropertyGraph.h [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+/// uninterpreted and can be manipulated as a collection of bits. A
+/// TypedPropertyGraph is appropriate for cases where computation needs to be done
+/// on the properties themselves.
+///
+/// \tparam NodeProps A tuple of property types (\ref Properties.h) for nodes
+/// \tparam EdgeProps A tuple of property types for edges
+template <typename NodeProps, typename EdgeProps>
+class TypedPropertyGraph {
+  using NodeView = PropertyViewTuple<NodeProps>;
+  using EdgeView = PropertyViewTuple<EdgeProps>;
```
Comment (witchel) [inline libgalois/include/katana/TypedPropertyGraph.h:35]: Oh, TypedPropertyGraph, ya mae be foolin' the others with yer fancy name and yer new closes, but I recognize those template arguments as the mark of the Devil! (actually, I thought this was DistGraph, but it is the old PropertyGraph, which I never used)

**Review by witchel — APPROVED** [review]
Naming is thinking, so thank you!
It is cool how you can update the variable names too.  I think I named a function PFGCreate, but those can be weeded out by hand.

File: libgalois/include/katana/TypedPropertyGraph.h (hunk 3 above)
Comment (ddn0) [reply libgalois/include/katana/TypedPropertyGraph.h:35]: TypedPropertyGraph nee PropertyGraph is mainly used in analytics applications themselves, and by the great spirit of Alan Turing, shouldn't find much use within the internal infrastructure of katana itself.

**Review by roshandathathri — APPROVED** [review]
LGTM

**Review by gurbinder533 — APPROVED** [review]
Looks great.

**Pull request merged**